### ISSUE TITLE
HBS Smart Contract

### ISSUE THREAD

**Issue by Mikerah** [initc3/MP-SPDZ#2]
For HoneyBadgerSWap, the smart contract needs the following minimal functionality:

- [ ] : Take inputs (as input masks) from clients in batches of x mins. Note: Minimize x based on activity of the chain
- [ ] : MPC nodes register in order to use Ethereum as a coordinator and for exporting the results of the AMM computation on-chain (in secret-shared form) for the clients to query Ethereum for this outputs.
- [ ] : Keeps track of deposits and withdrawals of the clients 

TLDR: Fork `asynchromix.sol` and change some parts for HBS.

**Comment by Mikerah** [comment]
@lilione finished this.
